tools/autograd/gen_variable_type.py

```diff
@@ -598,24 +615,15 @@ def setup_derivative():
         body.append(SETUP_DERIVATIVE.substitute(env, setup=setup))
         return body
 
-    def find_args_with_derivatives():
-        """Find arguments that have derivative definitions"""
-        if func is None:
-            return differentiable_inputs
-        names = set(name for d in func['derivatives'] for name in d['var_names'])
-        differentiable = [arg for arg in differentiable_inputs if arg['name'] in names]
-        if len(differentiable) != len(names):
-            missing = names - set(arg['name'] for arg in differentiable)
-            raise RuntimeError('Missing arguments for derivatives: {} in {}'.format(missing, func['name']))
-        return differentiable
-
     def emit_check_no_requires_grad(tensor_args, args_with_derivatives):
         """Checks that arguments without derivatives don't require grad"""
         body = []
         for arg in tensor_args:
             if arg in args_with_derivatives:
                 continue
             name = arg['name']
+            if name in args_with_no_gradients:
```

Comment:
this is super confusing because of the non-congruence of args_with_derivatives (full arguments) and args_with_no-gradients (just names).  I'm not sure why this doesn't use args_with_gradients, but I guess this was an issue before your change...

Reply:
Yes, variable_type uses derivatives, whereas load_derivatives uses gradients. I'll just change gradients to derivatives.